webapp/src/app.tsx

```diff
@@ -1719,12 +1721,15 @@ export class ProjectView
             pxt.log("Tried to start simulator when we shouldn't");
             return;
         }
+        if (this.runToken) this.runToken.cancel()
+        this.runToken = new pxt.Util.CancellationToken()
```

Comment:
I think you need startOperation?

Reply:
I'm only using the token to detect cancel.